protos/authorization.proto

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// -----------------------------------------------------------------------------
+
+syntax = "proto3";
+option java_multiple_files = true;
+option java_package = "sawtooth.sdk.protobuf";
+option go_package = "authorization_pb2";
+
+message ConnectionRequest {
+	string endpoint = 1;
+}
+
+enum RoleType {
+	// A shorthand request for asking for all allowed roles.
+	ALL = 0;
+	// Role defining validator to validator communication
+  NETWORK = 1;
+}
+
+message ConnectionResponse {
+  enum Status {
+      OK = 0;
+      ERROR = 1;
+  }
+
+	enum RoleType {
```

Comment:
Does this enum need to be repeated in all these message definitions?

Reply:
I was having problems making it work when it was defined outside but i think i figured it out and will update